tests/node/test_wakunode_filter.nim

```diff
@@ -107,3 +107,117 @@ suite "Full Node - Waku Filter - End to End":
     # Then the message is not pushed to the client
     check:
       not await pushHandlerFuture.withTimeout(FUTURE_TIMEOUT)
+  
+  asyncTest "Client Node can't receive Push from Server Node, via Relay":
+    # Given the server node has Relay enabled
+    await server.mountRelay()
+
+    # And valid filter subscription
+    let subscribeResponse = await client.filterSubscribe(
+      some(pubsubTopic), contentTopicSeq, serverRemotePeerInfo
+    )
+    require:
+      subscribeResponse.isOk()
+      server.wakuFilter.subscriptions.len == 1
+
+    # When a server node gets a Relay message
+    let msg1 = fakeWakuMessage(contentTopic=contentTopic)
+    await server.publish(some(pubsubTopic), msg1)
+
+    # Then the message is not sent to the client's filter push handler
+    check (not await pushHandlerFuture.withTimeout(FUTURE_TIMEOUT))
+
+  asyncTest "Client Node can't subscribe to Server Node without Filter":
+    # Given a server node with Relay without Filter
+    let
+      serverKey = generateSecp256k1Key()
+      server = newTestWakuNode(serverKey, ValidIpAddress.init("0.0.0.0"), Port(0))
+    
+    waitFor server.start()
+    waitFor server.mountRelay()
+
+    let serverRemotePeerInfo = server.peerInfo.toRemotePeerInfo()
+
+    # When a client node subscribes to the server node
+    let subscribeResponse = await client.filterSubscribe(
+      some(pubsubTopic), contentTopicSeq, serverRemotePeerInfo
+    )
+
+    # Then the subscription is successful
+    check (not subscribeResponse.isOk())
+
+  asyncTest "Filter Client Node can receive messages after subscribing and stopping without unsubscribing, via Filter":    
+    # Given a valid filter subscription
+    let subscribeResponse = await client.filterSubscribe(
+      some(pubsubTopic), contentTopicSeq, serverRemotePeerInfo
+    )
+    require:
+      subscribeResponse.isOk()
+      server.wakuFilter.subscriptions.len == 1
+    
+    # And the client node reboots
+    waitFor client.stop()
+    waitFor client.start()
+    client.mountFilterClient()
+
+    # When a message is sent to the subscribed content topic, via Filter; without refreshing the subscription
+    let msg1 = fakeWakuMessage(contentTopic=contentTopic)
+    await server.filterHandleMessage(pubsubTopic, msg1)
+
+    # Then the message is not pushed to the client
+    check (not await pushHandlerFuture.withTimeout(FUTURE_TIMEOUT))
+
+    # Given the client refreshes the subscription
+    # TODO: CHECK IF THIS IS NECESSARY. 
+    # AT FIRST GLANCE IT SEEMS TO COLLIDE WITH WAKU_FILTER_CLIENT'S BEHAVIOUR: SHOULD NOT NEED?
+    let subscribeResponse2 = await client.filterSubscribe(
```

Comment:
Is this required for the test to pass? I would expect that it shouldn't be needed as long as:
1. the service node didn't restart
2. the client starts up again with the same peerId and transport as before

Reply:
I think so too, even more given the client doesn't do any unsubscribing, apparently. I'll open a ticket.